src/components/PopupWithForm.js

```diff
@@ -22,11 +39,14 @@ export default class PopupWithForm extends Popup {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
+      this.showLoading();
+      this._handleFormSubmit(this._getInputValues()).finally(() =>
+        this.hideLoading()
```

Comment:
Needs correction: you should end the loading in index.js in finally, because the form can be submitted using different ways, not exactly with the apu request

This class should not know anything about this._handleFormSubmit logic